web/src/app/admin/admin-api-keys/AdminAPIKeysDeleteDialog.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from 'react'
+import { GQLAPIKey } from '../../../schema'
+import Button from '@mui/material/Button'
+import Dialog from '@mui/material/Dialog'
+import DialogActions from '@mui/material/DialogActions'
+import DialogContent from '@mui/material/DialogContent'
+import DialogContentText from '@mui/material/DialogContentText'
+import DialogTitle from '@mui/material/DialogTitle'
+import { gql, useMutation } from '@apollo/client'
+import { GenericError } from '../../error-pages'
+import Spinner from '../../loading/components/Spinner'
+
+// query for deleting API Key which accepts API Key ID
+const deleteGQLAPIKeyQuery = gql`
+  mutation DeleteGQLAPIKey($id: ID!) {
+    deleteGQLAPIKey(id: $id)
+  }
+`
+
+export default function AdminAPIKeysDeleteDialog(props: {
+  apiKey: GQLAPIKey | null
```

Comment:
Rather than passing in the entire API key, let's just ask for the ID and then have the dialog itself grab the key info. Add a query here to fetch the keys and grab the one we need to display it's name in the dialog.

Use `PolicyStepDeleteDialog` as an example of how to do this.

Additionally:
- Use `<FormDialog>` and ensure to pass along the possible mutation error (so the retry button works)
- Make sure to set `additionalTypenames: ['GQLAPIKey']` so the app reloads data properly

Reply:
I think the gqlAPIKeys query needs to be updates to accepts GQLAPIKey input before I could integrate fetching by ID, is that right?